Makefile

```diff
@@ -386,3 +395,23 @@ verify-generated: bundle generate
 
 install-git-hooks:
 	git config core.hooksPath .githooks	
+
+
+GOLANGCI_LINT_LOCAL_VERSION := $(shell command ${GOLANGCI_LINT} --version 2> /dev/null | awk '{print $$4}')
+.PHONY: golangci-lint
+golangci-lint:
```

Comment:
Can we add this target as prerequisite  for the `build` target?

Reply:
Tie both targets in Makefile will not allow us to run a build without running this check and I think we could wanna do that on some environments.

The idea is to run them automatically using a prow job once this PR is merge, but keep that job as `optional` (non-blocking for merge) in the beginning until we feel confident enough to change it.

If running this in prow is "too late" and the problem is to avoid forget running this before pushing changes, I think a git hook could do the tick.

WDYT?

Reply:
> If running this in prow is "too late" and the problem is to avoid forget running this before pushing changes, I think a git hook could do the tick.

yes that could work indeed. Thanks.

Reply:
I don't mind this approach but I'd start with an informing (non-blocking) CI lane and "enforce" the fixes during the review.

Reply:
I was pretending to do that with a prow ci-lane but with the gh action I think I do not have enough privileges to mark the action as "non-required"

Reply:
In our case prow considers all GH actions lane as non-binding. Admittedly this is incidental configuration, but good enough for now.